### PR TITLE
Feature/support binary files

### PR DESCRIPTION
**Pull request by rmelian**
adding support for ignore files on templates

### CODE REVIEW

File: templates/java-spring/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (fmvilas) [inline templates/java-spring/config.json]: Use another name for the file. `config.json` can easily collide with some project template needs. I'd suggest something like `.ag-config.json`.

File: lib/generator.js [hunk 2]
```diff
@@ -50,22 +51,34 @@ const renderFile = (filePath, data) => new Promise((resolve, reject) => {
...
   const target_dir = options.target_dir;
   const file_name = options.file_name;
   const root = options.root;
   const data = options.data;
 
-  renderFile(path.resolve(root, file_name), data)
-    .then((parsed_content) => {
-      const template_path = path.relative(templates_dir, path.resolve(root, file_name));
-      const generated_path = path.resolve(target_dir, template_path);
+  if (isFileIgnored(file_name, template_config.ignore)) {
```
Comment (fmvilas) [inline lib/generator.js]: Maybe we should come up with a better name here and in the config file? Ignoring sounds like it shouldn't be even copied.

File: templates/java-spring/config.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+  "ignore": [
```
Comment (fmvilas) [inline templates/java-spring/config.json]: Ignore sounds like something different. Shall we use another name?

File: templates/java-spring/config.json (hunk 1 above)
Comment (rmelian) [reply templates/java-spring/config.json]: what about `.tp-config.json`?

Comment (fmvilas) [reply templates/java-spring/config.json]: That sounds good. Whatever is not "easy to collide with" 😊

**Pull request merged**